src/Abp.EntityFrameworkCore.EFPlus/EntityFrameworkCore/EFPlus/AbpEntityFrameworkCoreEFPlusExtensions.cs

```diff
@@ -0,0 +1,157 @@
+        public static async Task<int> BatchDeleteAsync<TEntity>([NotNull] this IRepository<TEntity> repository, Expression<Func<TEntity, bool>> predicate)
+            where TEntity : Entity<int>
+        {
+            return await repository.BatchDeleteAsync<TEntity, int>(predicate);
+        }
+
+        /// <summary>
+        /// Updates all matching entities using given updateExpression for given predicate
+        /// </summary>
+        /// <typeparam name="TEntity">Entity type</typeparam>
+        /// <typeparam name="TPrimaryKey">Primary key type</typeparam>
+        /// <param name="repository">Repository</param>
+        /// /// <param name="updateExpression">Update expression</param>
+        /// <param name="predicate">Predicate to filter entities</param>
+        /// <returns></returns>
+        public static async Task<int> BatchUpdateAsync<TEntity, TPrimaryKey>([NotNull]this IRepository<TEntity, TPrimaryKey> repository, Expression<Func<TEntity, TEntity>> updateExpression, Expression<Func<TEntity, bool>> predicate)
+            where TEntity : Entity<TPrimaryKey>
+        {
+            Check.NotNull(repository, nameof(repository));
+
+            var query = repository.GetAll().IgnoreQueryFilters();
+
+            var abpFilterExpression = GetFilterExpressionOrNull<TEntity, TPrimaryKey>(repository.GetIocResolver());
+            var filterExpression = ExpressionCombiner.Combine(predicate, abpFilterExpression);
+
+            if (filterExpression != null)
+            {
+                query = query.Where(filterExpression);
+            }
+
+            return await query.UpdateAsync(updateExpression);
+        }
+
+        /// <summary>
+        /// Updates all matching entities using given updateExpression for given predicate
+        /// </summary>
+        /// <typeparam name="TEntity">Entity type</typeparam>
+        /// <param name="repository">Repository</param>
+        /// /// <param name="updateExpression">Update expression</param>
+        /// <param name="predicate">Predicate to filter entities</param>
+        /// <returns></returns>
+        public static async Task<int> BatchUpdateAsync<TEntity>(
+            this IRepository<TEntity> repository, Expression<Func<TEntity, TEntity>> updateExpression,
+            Expression<Func<TEntity, bool>> predicate)
+            where TEntity : Entity<int>
+        {
+            return await repository.BatchUpdateAsync<TEntity, int>(updateExpression, predicate);
+        }
+
+        private static Expression<Func<TEntity, bool>> GetFilterExpressionOrNull<TEntity, TPrimaryKey>(IIocResolver iocResolver) where TEntity : Entity<TPrimaryKey>
+        {
+            Expression<Func<TEntity, bool>> expression = null;
+
+            if (typeof(ISoftDelete).IsAssignableFrom(typeof(TEntity)))
+            {
+                var isSoftDeleteFilterEnabled = iocResolver.Resolve<ICurrentUnitOfWorkProvider>().Current?.IsFilterEnabled(AbpDataFilters.SoftDelete) == true;
```

Comment:
Must release the resolved `ICurrentUnitOfWorkProvider`, otherwise it causes to memory leaks. You can use iocResolver.CreateScope for this method body and resolve from it. This is also valid for the GetCurrentTenantIdOrNull method.

Also, it is done 3 times, I suggest to resolve it and use everywhere.